internal/third_party/golang/expansion/expand.go

```diff
@@ -19,23 +19,23 @@ func syntaxWrap(input string) string {
 // implements the expansion semantics defined in the expansion spec; it
 // returns the input string wrapped in the expansion syntax if no mapping
 // for the input is found.
-func MappingFuncFor(context ...map[string]string) func(string) string {
-	return func(input string) string {
+func MappingFuncFor(context ...map[string]string) func(string) (string, bool) {
```

Comment:
Since this file is in the `third_party` directory, it was likely copy-pasted. It might be helpful to add a brief comment explaining any modifications to ensure we remember the changes.

Reply:
I think that a top-level comment near the beginning of the file would be helpful. It could say something like: "This is a third-party file, but we modified the `MappingFuncFor` and `Expand` functions to return a second parameter indicating whether the environments were found".